vnext/ReactUWP/Modules/DeviceInfoModule.cpp

```diff
@@ -20,25 +20,35 @@ DeviceInfo::DeviceInfo(const std::shared_ptr<IReactInstance> &reactInstance) : m
 
 void DeviceInfo::update() {
   auto displayInfo = winrt::Windows::Graphics::Display::DisplayInformation::GetForCurrentView();
+  m_scale = static_cast<float>(displayInfo.ResolutionScale()) / 100;
   winrt::Windows::UI::ViewManagement::UISettings uiSettings;
 
   auto const &window = winrt::Windows::UI::Xaml::Window::Current().CoreWindow();
-
+  m_width = window.Bounds().Width;
+  m_height = window.Bounds().Height;
   m_dimensions = folly::dynamic::object(
       "windowPhysicalPixels",
-      folly::dynamic::object("width", window.Bounds().Width)("height", window.Bounds().Height)(
-          "scale", static_cast<int>(displayInfo.ResolutionScale()) / 100)("fontScale", uiSettings.TextScaleFactor())(
-          "densityDpi", displayInfo.LogicalDpi()))(
+      folly::dynamic::object("width", m_width * m_scale)("height", m_height * m_scale)(
+          "scale", m_scale)("fontScale", uiSettings.TextScaleFactor())("densityDpi", displayInfo.LogicalDpi()))(
       "screenPhysicalPixels",
       folly::dynamic::object("width", displayInfo.ScreenWidthInRawPixels())(
-          "height", displayInfo.ScreenHeightInRawPixels())(
-          "scale", static_cast<int>(displayInfo.ResolutionScale()) / 100)("fontScale", uiSettings.TextScaleFactor())(
+          "height", displayInfo.ScreenHeightInRawPixels())("scale", m_scale)("fontScale", uiSettings.TextScaleFactor())(
           "densityDpi", displayInfo.LogicalDpi()));
+
+    m_dpiChangedRevoker = displayInfo.DpiChanged(winrt::auto_revoke, [this](const auto &, const auto &) {
+        auto displayInfo = winrt::Windows::Graphics::Display::DisplayInformation::GetForCurrentView();
+        m_scale = static_cast<float>(displayInfo.ResolutionScale()) / 100;
+        m_dimensions["windowPhysicalPixels"]["scale"] = m_scale;
+        m_dimensions["windowPhysicalPixels"]["densityDpi"] = displayInfo.LogicalDpi();
+        m_dimensions["screenPhysicalPixels"]["scale"] = m_scale;
+        m_dimensions["screenPhysicalPixels"]["densityDpi"] = displayInfo.LogicalDpi();
+        updateRootElementSize(m_width, m_height);
+  });
 }
 
 void DeviceInfo::updateRootElementSize(float width, float height) {
-  m_dimensions["windowPhysicalPixels"]["width"] = width;
-  m_dimensions["windowPhysicalPixels"]["height"] = height;
+  m_dimensions["windowPhysicalPixels"]["width"] = width * m_scale;
+  m_dimensions["windowPhysicalPixels"]["height"] = height * m_scale;
```

Comment:
I'm confused by the intent of this code.  The property is "windowPhysicalPixels" but the value we're sending is not the window size, but rather the size of the "root" XAML element.  This element can be any size, depending on how the app is authored.  Why the root and not the window?  And how does the app learn about the actual window size?

Related question:  If an app contains more than one ReactControl, will this code be called for all roots?  If so that seems potentially like a bug as each root can be a different size.

My instinct here is that this API should reflect the size of the window, but I'm open to discussion.  Note:  that has implications on XAML islands, which might not have a window this code knows about, and would need further discussion.

Reply:
We switched to root element's size instead of window size in PR for #2470. 
Using root element size to represent "windowPhyscialPixels" makes more sense for RNW as it is not targeted for full window based phone app.
- If the RN root is also the root of the window, root element size is the same as window size.
- If the RN root is not root of the window, returning window size as dimension has negative effect as RN core use dimension for layout process, and we are telling RNCore of dimension which are bigger than its actual size.
- Similarly, if app has mutliple RN root, each RN instance should have its own "dimension".
- Is should also work for the island case, but there maybe minor tweak to call Island API instead of CoreWindow API.


---
In reply to: [371593661](https://github.com/microsoft/react-native-windows/pull/3956#discussion_r371593661) [](ancestors = 371593661)

Reply:
I'm actually not sure if it's possible to share the DeviceInfoModule between root views.  I posted a question on the Teams channel.

---
In reply to: [371958464](https://github.com/microsoft/react-native-windows/pull/3956#discussion_r371958464) [](ancestors = 371958464,371593661)

Reply:
After discussions and investigation on current Android and iOS implementation, I am switching over "WindowsPhysicalPixels" to CoreWindow's width and height.

---
In reply to: [372052393](https://github.com/microsoft/react-native-windows/pull/3956#discussion_r372052393) [](ancestors = 372052393,371958464,371593661)

Reply:
In that case we should remove updateRootElementSize() altogether.  I don't see any code listening for CoreWindow size changes either?

---
In reply to: [373223200](https://github.com/microsoft/react-native-windows/pull/3956#discussion_r373223200) [](ancestors = 373223200,372052393,371958464,371593661)

Reply:
Much cleaner now thank you!

---
In reply to: [373228761](https://github.com/microsoft/react-native-windows/pull/3956#discussion_r373228761) [](ancestors = 373228761,373223200,372052393,371958464,371593661)